docs/src/stake-pool.md

```diff
@@ -1,11 +1,427 @@
+
+```sh
+$ spl-stake-pool withdraw --pool 3CLwo9CntMi4D1enHEFBe3pRJQzGJBCAYe66xFuEbmhC  --amount 0.02 --burn-from 34XMHa3JUPv46ftU4dGHvemZ9oKVjnciRePYMcX3rjEF
+Withdrawing from account FYQB64aEzSmECvnG8RVvdAXBxRnzrLvcA3R22aGH2hUN, amount 8.867176821 SOL, 0.019999999 pool tokens
+Creating account to receive stake CZF2z3JJoDmJRcVjtsrz1BKUUGNL3VPW5FPFqge1bzmQ
+Signature: 2xBPVPJ749AE4hHNCNYdjuHv1EdMvxm9uvvraWfTA7Urrvecwh9w64URCyLLroLQ2RKDGE2QELM2ZHd8qRkjavJM
+```
+
+Our 0.02 pool tokens were taken, and in exchange we received a fully active stake
+account, delegated to `2HUKQz7W2nXZSwrdX5RkfS2rLU4j1QZLjdGCHcoUKFh3`.  We can
+double-check the status of the stake account:
+
+```sh
+$ solana stake-account CZF2z3JJoDmJRcVjtsrz1BKUUGNL3VPW5FPFqge1bzmQ
+Balance: 8.869459257 SOL
+Rent Exempt Reserve: 0.00228288 SOL
+Delegated Stake: 8.867176377 SOL
+Active Stake: 8.867176377 SOL
+Delegated Vote Account Address: 2HUKQz7W2nXZSwrdX5RkfS2rLU4j1QZLjdGCHcoUKFh3
+Stake Authority: 4SnSuUtJGKvk2GYpBwmEsWG53zTurVM8yXGsoiZQyMJn
+Withdraw Authority: 4SnSuUtJGKvk2GYpBwmEsWG53zTurVM8yXGsoiZQyMJn
+```
+
+Note: due to truncation on SPL token to SOL conversion, we will not receive the
+exact amount calculated by the command-line utility.
+
+We can leave this stake account as it is, or we can deactivate it as a normal
+stake account.
+
+```sh
+$ solana deactivate-stake CZF2z3JJoDmJRcVjtsrz1BKUUGNL3VPW5FPFqge1bzmQ
+Signature: 816LjrZFPWcdvkeFTBezFxptMoNjUEyVmHQ6bpX7sdZZbzhWKchUyx2MC5VHdXFP7U41B8S7QQfb7hTqDASCqZg
+```
+
+Once the stake is no longer active, we can use it as normal SOL.
+
+### Example: (Administrator only) Remove validator stake account
+
+If the stake pool manager wants to stop delegating to a vote account, they can
+totally remove the validator stake account from the stake pool.
+
+```sh
+$ spl-stake-pool remove-validator-stake --pool 3CLwo9CntMi4D1enHEFBe3pRJQzGJBCAYe66xFuEbmhC --stake CrStLEWfme37kDc3nubK9HsmWR5dsuVUuqEKqTR4Mc5E --burn-from 34XMHa3JUPv46ftU4dGHvemZ9oKVjnciRePYMcX3rjEF
+Signature: 5rrQ3xhDWyiPkUTAQkNAeq31n6sMf1xsg2x9hVY8Vj1NonwBnhxuTv87nADLkwC8Xzc4CGTNCTX2Vph9esWnXk2d
+```
+
+This operation works just like `withdraw`, in that the stake pool manager provides
+staking derivatives in exchange for an activated stake account. The difference
+is that the validator stake account is totally removed from the stake pool and
+now belongs to the administrator.
+
+We can check the removed stake account:
+
+```sh
+$ solana stake-account CrStLEWfme37kDc3nubK9HsmWR5dsuVUuqEKqTR4Mc5E
+Balance: 1.002282881 SOL
+Rent Exempt Reserve: 0.00228288 SOL
+Delegated Stake: 1.000000001 SOL
+Active Stake: 1.000000001 SOL
+Delegated Vote Account Address: AUCzCaGAGjL3uyjFBtJs7KuJcgQWvNZu1Z2S9G3pw77G
+Stake Authority: 4SnSuUtJGKvk2GYpBwmEsWG53zTurVM8yXGsoiZQyMJn
+Withdraw Authority: 4SnSuUtJGKvk2GYpBwmEsWG53zTurVM8yXGsoiZQyMJn
+```
+
+The SPL token account has been debited to accommodate the removal of SOL from
+the stake pool.
```

Comment:
Maybe this section could be expanded on.  It's not immediately clear to me how the system prevents a malicious administrator from running `spl-stake-pool remove-validator-stake` on all the stake pools and running away with the SOL

Reply:
They have to go through the same mechanism as outside users and provide pool tokens in exchange for staked SOL.  Admins can add / remove validator accounts totally though.  I'll clarify that